packages/blocks/src/__internal__/utils/selection.ts

```diff
@@ -93,16 +93,31 @@ export function focusRichText(
     const { x, y } = position;
     let newTop = y;
     let newLeft = x;
-    if (oldBottom <= y) {
-      newTop = bottom - lineHeight / 2;
+    if (bottom <= y) {
+      let finalBottom = bottom;
+      if (bottom < 100 && scrollContainer) {
```

Comment:
Is the `100` here a magic number? If so we can make it a `const` since looks like it repeats for multi times.